### ISSUE TITLE
no-unused-prop-types cannot detect unused property

### ISSUE THREAD

**Issue by victor-homyakov** [yannickcr/eslint-plugin-react#880]
``` jsx
const Comp = (props) => <div className={props.nested.className}>{props.nested.text}</div>;
Comp.propTypes = {
    nested: React.PropTypes.object.isRequired,
    text: React.PropTypes.string.isRequired
};
```

Property `text` should be marked as unused.
Rule configuration: `"react/no-unused-prop-types": ["error", {skipShapeProps: true}]`.

**Comment by EvNaverniouk** [comment]
In your example code base `text` is **not** used. So the error is displayed correctly. Your component uses `props.nested.text` but your have a PropTypes defined for `props.text`.

**Comment by victor-homyakov** [comment]
Please read description one more time. Property `text` is *not used*, but is *not* flagged as unused.

**Comment by ljharb** [comment]
It's only fixable for the modern TS eslint parser; the old one fails to provide the information required.
